README.md

```diff
@@ -1,170 +1,60 @@
-
-```lua
-local lspconfig = require'lspconfig'
-local configs = require'lspconfig/configs'
--- Check if it's already defined for when reloading this file.
-if not lspconfig.foo_lsp then
-  configs.foo_lsp = {
-    default_config = {
-      cmd = {'/home/ashkan/works/3rd/lua-language-server/run.sh'};
-      filetypes = {'lua'};
-      root_dir = function(fname)
-        return lspconfig.util.find_git_ancestor(fname) or vim.loop.os_homedir()
-      end;
-      settings = {};
-    };
-  }
-end
-lspconfig.foo_lsp.setup{}
-```
+Enabling most language servers is as easy as:
 
-### Example: override default config for all servers
-
-If you want to change default configs for all servers, you can override default_config like this. In this example, we additionally add a check for log_level and message_level which can be passed to the server to control the verbosity of "window/logMessage".
-
-```lua
-local lspconfig = require'lspconfig'
-lspconfig.util.default_config = vim.tbl_extend(
-  "force",
-  lspconfig.util.default_config,
-  {
-    autostart = false,
-    handlers = {
-      ["window/logMessage"] = function(err, method, params, client_id)
-          if params and params.type <= vim.lsp.protocol.MessageType.Log then
-            vim.lsp.handlers["window/logMessage"](err, method, params, client_id)
-          end
-        end;
-      ["window/showMessage"] = function(err, method, params, client_id)
-          if params and params.type <= vim.lsp.protocol.MessageType.Warning.Error then
-            vim.lsp.handlers["window/showMessage"](err, method, params, client_id)
-          end
-        end;
-    }
-  }
-)
+```vim
+lua << EOF
+require'lspconfig'.rust_analyzer.setup{}
+EOF
 ```
-
-## Individual server settings and initialization options
-
-See [CONFIG.md](CONFIG.md) for documentation and configuration of individual language servers.
-This document contains installation instructions for each language server, and is 
-auto-generated from the documentation in the lua source. Do not submit PRs 
-modifying CONFIG.md directly; CONFIG.md will be overwritten by docgen
-
-**You do not need to copy settings or init_options from this configuration into your config**
+For a full list of servers, see [CONFIG.md](CONFIG.md). This document contains installation instructions and additional, optional customization suggestions for each language server. For some servers that are not on your system path (jdtls, elixirls) you will be required to manually add `cmd` as an entry in the table passed to setup.
 
 ## Keybindings and completion
 
-The following maps most of the standard functions to keybindings, and maps omnicomplete to use
-the lsp.omnifunc. See `:help lsp` for more details
+nvim-lspconfig does not include any keybindings or completion enabled by default. The following snippet uses the on-attach function to map keybindings and omnifunc (\<c-x\>\<c-o\>) for completion whenever the language server client activates on a given buffer:
```

Comment:
I think there is an educational point to be made here, I suspect a lot of users coming from coc for example don't understand the autocompletion features are being provided by coc, and so when they come here, they don't realize either they need to learn how to trigger completion via omnifunc's binding <c-x><c-o> and hook up omnifunc via `buf_set_option('omnifunc', 'v:lua.vim.lsp.omnifunc')` as below, or that they need to install another plugin, e.g. deoplete, or compe etc and then configure that. So while this aids the user in understanding that it doesn't include keybindings and complete by default, it doesn't help them to understand what to expect from lspconfig after pasting the below.

Reply:
What do you mean? I do include the keybinding in the above for omnifunc snippet for omnifunc, and it is added in the example configurations + keybinding. What would you suggest we add to clarify?

 Totally agree re: the confusion about autocompletion vs completion...

Reply:
I think I just mean to emphasize that a user who wants auto-completion (rather than just completion) will need to explore plugins that specialize in adding such features to nvim.

Something like the following, though I acknowledge this is more long winded:

> nvim-lspconfig does not automatically add keybindings (e.g. for go to definition, hover) or configure nvim's inbuilt omnifunc completion by default. If auto-completion beyond nvim's inbuilt completion is desired another plugin is needed. However, the following example snippet demonstrates how to provide an on-attach function in order to map keybindings and nvim's omnifunc (\<c-x\>\<c-o\>) for completion whenever the language server client activates on a given buffer:

Reply:
I also think it's on us here to educate a user about what an on_attach function and even what buffer local mappings are.

Something like, again I understand it is long winded:

Given that we don't want lsp related keybindings added to all buffers, only buffers in which a server is active, we take advantage of the on_attach property when configuring each server. This function, when provided, will be executed when a server auto attaches for an appropriate buffer, meaning you will now have buffer local mappings provided as well as the appropriate configuration of omnifunc.

Reply:
I think I worked those both in, I tried to reduce the word count a bit but it's in the prelude to the example keybindings + completion

Reply:
Sounds good